integration-tests/test_registration.py

```diff
@@ -93,3 +98,38 @@ def test_register_group_option(insights_client, legacy_upload_value):
         check=False,
     )
     assert register_group_option.returncode == 0
+
+def test_machineid_changes_when_subman_is_unregistered_and_registered(insights_client, subman, test_config):
+    """machine-id should be different based on the auth method"""
+    if 'satellite' in test_config.environment:
+        pytest.skip("this test requires BASIC authentication")
+
+    insights_client.unregister()
+    assert not insights_client.is_registered
+    insights_client.register()
+    with open(MACHINE_ID_FILE, "r") as f:
+        machine_id_cert = f.read()
+    insights_client.unregister()
+
+    insights_client.config.auto_config = False
+    insights_client.config.authmethod = "BASIC"
+    insights_client.config.username = test_config.get("candlepin", "username")
+    insights_client.config.password = test_config.get("candlepin", "password")
+    insights_client.config.save()
+
+    insights_client.register()
+    with open(MACHINE_ID_FILE, "r") as f:
+        machine_id_random = f.read()
+
+    assert machine_id_random != machine_id_cert
+
+
+def test_registered_and_unregistered_files_are_created_and_deleted(insights_client):
+    """""'.registered and .unregistered file gets created and deleted"""
+    assert not insights_client.is_registered
+
```

Comment:
I would suggest to assert here that .registered doesn't exist before registration and similiar assertion for .unregister before unregistration